### DISCUSSION TITLE
Remove vault dependency in fybrik

### DISCUSSION THREAD

**Discussion by lappanna** [fybrik/fybrik#810] [Ideas]
We want fybrik to work without vault

**Comment by simanadler** [comment]
@lappanna Vault is used to manage (a) data set credentials, (b) user credentials for the external systems accessed - data catalog, policy manager, credential manager, (c) infrastructure credentials - internally used data stores (ex: for temporary copies, storing metrics, etc).  When you say that you want fybrik to work without vault do you mean you want it to be configured to potentially use an existing external vault instance rather than installing its own, or are you saying that you want to remove these capabilities from fybrik?

**Comment by lappanna** [comment]
we want fybrik configured to use external vault service.

**Reply by revit13** [reply]
@lappanna [Vault plugins](https://www.vaultproject.io/docs/internals/plugins) are used to retrieve credentials. To support it with external vault service, the service should be configured to use the plugin: https://learn.hashicorp.com/tutorials/vault/plugin-backends#setup-vault. In addition, the vault service should be configured to gain modules permissions to get the dataset credentials via the plugin. Please see https://github.com/fybrik/fybrik/blob/master/charts/vault/env/dev/vault-single-cluster-values.yaml for details on the vault configuration for single cluster setup (https://github.com/fybrik/fybrik/blob/master/charts/vault/README.md).

**Comment by revit13** [comment]
Please note that in fybrik/values.yaml file vault token is [defined](https://github.com/fybrik/fybrik/blob/master/charts/fybrik/values.yaml#L73) but it is used only for testing - the fybrik manager currently does not access vault, only the modules, as described in the previous comment.
links that might be relevant: 
- https://fybrik.io/v0.4/concepts/vault_plugins
- https://github.com/fybrik/fybrik/issues/335

**Reply by roee88** [reply]
So in the current implementation it depends on the connector and module right? If the connector is coded to work without actually accessing vault and the module has other means to obtain credentials then the vault installation can be skipped. Is that correct?

**Reply by shlomitk1** [reply]
If there is no secret attached to FybrikApplication, the connector won't use vault plugin at all. If modules do not get credentials using vault, the vault installation can be skipped.

**Reply by lappanna** [reply]
Do we have other ways to pass the credentials to the connector without using vault?
